cron/src/tests/test_message_service.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from datetime import datetime
+from pytest import fixture
+from freezegun import freeze_time
+from notifications.message_service import MessageService
+from notifications.definitions import RecordInfo, BookInfo, BorrowerInfo
+
+
+class TestMessageService():
+    def test_compose_messages_senders(self, test_data):
+        sender = 'foo <foo@example.com>'
+        template = '{{recipient_surname}}, {{recipient_name}}'
+        service = MessageService(sender)
+        messages = list(service.compose_messages(template, test_data))
+
+        def recipient_match(message, recipient_name, recipient_surname):
+            return str(message.get_content()) == '{}, {}\n'.format(
+                recipient_surname,
+                recipient_name)
+
+        for m in messages:
+            logging.info(m)
+
+        assert len(messages) == 2
+        assert len([
+            m for m in messages
+            if recipient_match(
+                m, 'borrower_name_1', 'borrower_surname_1')]) == 1
+
+        assert len([
+            m for m in messages
+            if recipient_match(
+                m, 'borrower_name_2', 'borrower_surname_2')]) == 1
+
+    def test_compose_messages_from(self, test_data):
+        sender = 'foo <foo@example.com>'
+        service = MessageService(sender)
+        messages = list(service.compose_messages('', test_data))
+
+        assert messages[0]['From'] == 'foo <foo@example.com>'
+        assert messages[1]['From'] == 'foo <foo@example.com>'
+
+    def test_compose_messages_to(self, test_data):
+        sender = 'foo <foo@example.com>'
+        service = MessageService(sender)
+        messages = list(service.compose_messages('', test_data))
+
+        assert len([
+            m for m in messages if
+            m['To'] == 'borrower_surname_1 ' +
+                       'borrower_name_1 <borrower_email_1@example.com>']) == 1
+
+        assert len([
+            m for m in messages if
+            m['To'] == 'borrower_surname_2 ' +
+                       'borrower_name_2 <borrower_email_2@example.com>']) == 1
+
+    @freeze_time(datetime(2012, 5, 20))
+    def test_compose_messages_items(self, test_data):
+        sender = 'foo <foo@example.com>'
+        service = MessageService(sender)
+        template = '{{#items}}{{title}} {{days_left}} |{{/items}}'
+        messages = list(service.compose_messages(template, test_data))
+
+        for m in messages:
+            logging.info(m)
+
+        assert len([
+            m for m in messages if
+            str(m.get_content()) == 'book1 1 |book2 2 |\n']) == 1
+
+        assert len([
+            m for m in messages if
+            str(m.get_content()) == 'book3 3 |\n']) == 1
+
+
+@fixture()
```

Comment:
To the present day we have been storing fixtures in conftest.py, it would be great to follow this rule.

Reply:
Pytest documentation advise to use conftest.py if you want ot use same fixture from different test files. This is not a case, co i would rather have it as a internal part of test.

More info:
http://doc.pytest.org/en/latest/fixture.html#conftest-py-sharing-fixture-functions

Reply:
ok